diff --git a/mesa-src/meson.build b/mesa-src/meson.build
index 2c3397d6..9990d3ef 100644
--- a/mesa-src/meson.build
+++ b/mesa-src/meson.build
@@ -1304,7 +1304,7 @@ elif with_dri_i965 and with_shader_cache
 endif
 
 # Determine whether or not the rt library is needed for time functions
-if host_machine.system() == 'windows' or cc.has_function('clock_gettime')
+if host_machine.system() == 'windows' or host_machine.system() == 'darwin' or cc.has_function('clock_gettime')
   dep_clock = null_dep
 else
   dep_clock = cc.find_library('rt')
